### ISSUE TITLE
CI test flake? failed to setup network for sandbox; `cni0: file exists`

### ISSUE THREAD

**Issue by estesp** [containerd/cri#1024]
Opening this per discussion in Slack with @Random-Liu.

From time to time the CRI tests cause a random failure in containerd/containerd CI. Restarting CI usually works and the failure has no known trigger; sometimes many PRs go through CI with no failures, and sometimes it fails more often. (one example after a merge of a successful PR to master: https://travis-ci.org/containerd/containerd/jobs/480505949)

The CRI test log usually looks like this:
```
BeforeEach] [k8s.io] Security Context
  /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/framework/framework.go:50
[BeforeEach] [k8s.io] Security Context
  /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/security_context.go:52
[It] runtime should support ContainerPID
  /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/security_context.go:220
STEP: create podSandbox with PID set to container
Jan 16 18:15:18.720: INFO: Pid namespace is "CONTAINER"
STEP: create NamespaceOption podSandbox
E0116 18:15:31.438909   14622 remote_runtime.go:96] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to setup network for sandbox "413532d612ba284bcfb2acedc7c7f34b6ac00267c03d1343e72fd886fc36838c": failed to set bridge addr: could not add IP address to "cni0": file exists
Jan 16 18:15:31.439: INFO: Unexpected error occurred: rpc error: code = Unknown desc = failed to setup network for sandbox "413532d612ba284bcfb2acedc7c7f34b6ac00267c03d1343e72fd886fc36838c": failed to set bridge addr: could not add IP address to "cni0": file exists
[AfterEach] [k8s.io] Security Context
  /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/framework/framework.go:51
[AfterEach] [k8s.io] Security Context
  /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/security_context.go:57
• Failure in Spec Teardown (AfterEach) [12.721 seconds]
[k8s.io] Security Context
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/framework/framework.go:72
  NamespaceOption
  /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/security_context.go:72
    runtime should support ContainerPID [AfterEach]
    /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/security_context.go:220
    failed to create PodSandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "413532d612ba284bcfb2acedc7c7f34b6ac00267c03d1343e72fd886fc36838c": failed to set bridge addr: could not add IP address to "cni0": file exists
    Expected error:
        <*status.statusError | 0xc000321920>: {
            Code: 2,
            Message: "failed to setup network for sandbox \"413532d612ba284bcfb2acedc7c7f34b6ac00267c03d1343e72fd886fc36838c\": failed to set bridge addr: could not add IP address to \"cni0\": file exists",
            Details: nil,
        }
        rpc error: code = Unknown desc = failed to setup network for sandbox "413532d612ba284bcfb2acedc7c7f34b6ac00267c03d1343e72fd886fc36838c": failed to set bridge addr: could not add IP address to "cni0": file exists
    not to have occurred
    /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/framework/util.go:152
```

The containerd log entry related to the `cni0: file exists` usually just has this one line:
```
time="2019-01-16T18:15:31.431161983Z" level=error msg="RunPodSandbox for &PodSandboxMetadata{Name:NamespaceOption-PodSandbox-ae13df62-19ba-11e9-802e-42010a1402a5,Uid:cri-test-uidae1c2490-19ba-11e9-802e-42010a1402a5,Namespace:cri-test-namespaceae1c24ba-19ba-11e9-802e-42010a1402a5,Attempt:2,} failed, error" error="failed to setup network for sandbox \"413532d612ba284bcfb2acedc7c7f34b6ac00267c03d1343e72fd886fc36838c\": failed to set bridge addr: could not add IP address to \"cni0\": file exists"
```

**Comment by abhi** [comment]
This seems to be coming from netlink library used by the bridge cni plugin. We may need to may be flush the bridge cni0 before a CI run may be ? I am not sure if this is a CRI/containerd core issue.

**Comment by mikebrow** [comment]
Sounds like the cidr has been used.

**Comment by kzys** [comment]
@Random-Liu's fix on the plugins package has been merged and released as 0.7.6, and he has updated vendor.conf to consume the version (https://github.com/containerd/cri/commit/cda752fd59333d21e3de12e89efcdca800b8916c). So we are good now?

**Comment by Random-Liu** [comment]
@kzys I think so.

Feel free to reopen if this happens again.
